rollup/rollup_sync_service/rollup_sync_service.go

```diff
@@ -241,10 +241,15 @@ func (s *RollupSyncService) parseAndUpdateRollupEventLogs(logs []types.Log, endB
 				log.Warn("got nil when reading last finalized batch index. This should happen only once.")
 			}
 
+			parentBatchMeta := &rawdb.FinalizedBatchMeta{}
+			if startBatchIndex > 0 {
+				parentBatchMeta = rawdb.ReadFinalizedBatchMeta(s.db, startBatchIndex-1)
+			}
+
 			var highestFinalizedBlockNumber uint64
 			batchWriter := s.db.NewBatch()
 			for index := startBatchIndex; index <= batchIndex; index++ {
-				parentBatchMeta, chunks, err := s.getLocalInfoForBatch(index)
```

Comment:
So the issue was that we'd try to read it from DB and fail (since it's only written to DB after the for loop)?

Reply:
Yes.